apps/omg_child_chain/lib/omg_child_chain/fees/fee_updater.ex

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+defmodule OMG.ChildChain.Fees.FeeUpdater do
+  @moduledoc """
+  Decides whether fees will be updated from the fetched fees from the feed.
+  """
+
+  alias OMG.ChildChain.Fees.FeeMerger
+  alias OMG.Fees
+
+  @type feed_reading_t :: {pos_integer(), Fees.full_fee_t()}
+  @type can_update_result_t :: {:ok, feed_reading_t()} | :no_changes
+
+  @doc """
+  Newly fetched fees will be effective as long as the amount change on any token is significant
+  or the time passed from previous update exceeds the update interval.
+  """
+  @spec can_update(
+          stored_fees :: feed_reading_t(),
+          fetched_fees :: feed_reading_t(),
+          tolerance_percent :: pos_integer(),
+          update_interval_seconds :: pos_integer()
+        ) :: can_update_result_t()
+  def can_update({_, fee_spec}, {_, fee_spec}, _tolerance_percent, _update_interval_seconds), do: :no_changes
+
+  def can_update({t0, _}, {t1, _} = updated, _tolerance_percent, update_interval_seconds)
+      when t0 <= t1 and t1 - t0 >= update_interval_seconds,
+      do: {:ok, updated}
+
+  def can_update({_, stored_fees}, {_, fetched_fees} = updated, tolerance_percent, _update_interval_seconds) do
+    with merged when is_map(merged) <- merge_types(stored_fees, fetched_fees),
```

Comment:
we can avoid another with here it seems.
if `merge_types(stored_fees, fetched_fees)` returns :disjoint_types ... isn't that an error in the fee feed? why are we taking this as `:ok, updated`?

similarly for `false <- Enum.any?(merged, &token_mismatch?/1),`
if token missmatches, (returns true) this sounds like an error.

Reply:
Both cases are not errors, here `stored_fees` and `fetched_fees` are correct `fee_specs` but they differs on transaction type or a token within the same tx_type... which means we started / stoped to support tx_type or token which means the change is significant enough to update

Reply:
ok, if that's the case, `merge_fees` is doing too much atm. because it is not evident (from the name) what it's purpose is. try to break this down.
`token_mismatch ` might have a bad connotation. from the name, it implies it is either OK, or it is wrong (error).

Reply:
Refactored